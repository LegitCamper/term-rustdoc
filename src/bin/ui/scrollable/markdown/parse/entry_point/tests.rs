use super::{markdown_iter, parse};
use insta::{assert_debug_snapshot as snap, assert_display_snapshot as shot};

#[test]
fn parse_markdown() {
    let doc = r#"
# h1 `code`

aaa b *c* d **e**. ~xxx~ z.

1 *c **ss** d sadsad xxx* `yyyy`

```
let a = 1;
```

> rrr sss 
> tt

- [x] done!
    - nested list
- [ ] undone
    1. *a*
    2. `b`
"#;
    snap!(markdown_iter(doc).collect::<Vec<_>>());
    shot!(parse(doc), @r###"
    # h1 `code`

    aaa b c d e. xxx z.

    1 c ss d sadsad xxx `yyyy`

    let a = 1;

    rrr sss tt

    * [x] done!
      * nested list
    * [ ] undone
      1. a
      2. `b`

    "###);
}

#[test]
fn parse_markdown_links() {
    let doc = "
[a](b), [c], [e][c].

[c]: d

[`f`][c].

## h2 [c] [`h`][c]
";
    snap!(markdown_iter(doc).collect::<Vec<_>>());
    shot!(parse(doc), @r###"
    a, c, e.

    `f`.

    ## h2 c `h`

    "###);
}
