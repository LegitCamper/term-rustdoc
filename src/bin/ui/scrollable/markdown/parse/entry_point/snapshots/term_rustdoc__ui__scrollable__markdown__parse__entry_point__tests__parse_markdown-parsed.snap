---
source: src/bin/ui/scrollable/markdown/parse/entry_point/tests.rs
expression: blocks
---
Blocks {
    blocks: [
        Block {
            lines: [
                Line {
                    words: [
                        Word {
                            word: "# ",
                            style.add_modifier: BOLD,
                            tag: Heading(
                                0,
                            ),
                        },
                        Word {
                            word: "h1",
                            style.add_modifier: BOLD,
                            tag: Heading(
                                0,
                            ),
                            trailling_whitespace: true,
                        },
                        Word {
                            word: "`",
                            style.add_modifier: BOLD,
                            tag: Heading(
                                0,
                            ),
                        },
                        Word {
                            word: "code",
                            style.add_modifier: BOLD,
                            tag: Heading(
                                0,
                            ),
                        },
                        Word {
                            word: "`",
                            style.add_modifier: BOLD,
                            tag: Heading(
                                0,
                            ),
                        },
                    ],
                },
            ],
            links: [],
            footnotes: [],
        },
        Block {
            lines: [
                Line {
                    words: [
                        Word {
                            word: "aaa",
                            trailling_whitespace: true,
                        },
                        Word {
                            word: "b",
                            trailling_whitespace: true,
                        },
                        Word {
                            word: "c",
                            style.add_modifier: ITALIC,
                        },
                        Word {
                            word: "",
                            trailling_whitespace: true,
                        },
                        Word {
                            word: "d",
                            trailling_whitespace: true,
                        },
                        Word {
                            word: "e",
                            style.add_modifier: BOLD,
                        },
                        Word {
                            word: ".",
                            trailling_whitespace: true,
                        },
                        Word {
                            word: "xxx",
                            style.add_modifier: CROSSED_OUT,
                        },
                        Word {
                            word: "",
                            trailling_whitespace: true,
                        },
                        Word {
                            word: "z.",
                        },
                    ],
                },
            ],
            links: [],
            footnotes: [],
        },
        Block {
            lines: [
                Line {
                    words: [
                        Word {
                            word: "1",
                            trailling_whitespace: true,
                        },
                        Word {
                            word: "c",
                            style.add_modifier: ITALIC,
                            trailling_whitespace: true,
                        },
                        Word {
                            word: "ss",
                            style.add_modifier: ITALIC,
                        },
                        Word {
                            word: "",
                            style.add_modifier: ITALIC,
                            trailling_whitespace: true,
                        },
                        Word {
                            word: "d",
                            style.add_modifier: ITALIC,
                            trailling_whitespace: true,
                        },
                        Word {
                            word: "sadsad",
                            style.add_modifier: ITALIC,
                            trailling_whitespace: true,
                        },
                        Word {
                            word: "xxx",
                            style.add_modifier: ITALIC,
                        },
                        Word {
                            word: "",
                            trailling_whitespace: true,
                        },
                        Word {
                            word: "`",
                            tag: InlineCode,
                        },
                        Word {
                            word: "yyyy",
                            tag: InlineCode,
                        },
                        Word {
                            word: "`",
                            tag: InlineCode,
                        },
                    ],
                },
            ],
            links: [],
            footnotes: [],
        },
        Block {
            lines: [
                Line {
                    words: [
                        Word {
                            word: "let",
                            tag: CodeBlock(
                                "rust",
                            ),
                        },
                        Word {
                            word: " a ",
                            tag: CodeBlock(
                                "rust",
                            ),
                        },
                        Word {
                            word: "=",
                            tag: CodeBlock(
                                "rust",
                            ),
                        },
                        Word {
                            word: " ",
                            tag: CodeBlock(
                                "rust",
                            ),
                        },
                        Word {
                            word: "1",
                            tag: CodeBlock(
                                "rust",
                            ),
                        },
                        Word {
                            word: ";",
                            tag: CodeBlock(
                                "rust",
                            ),
                        },
                    ],
                },
            ],
            links: [],
            footnotes: [],
        },
        Block {
            lines: [
                Line {
                    words: [
                        Word {
                            word: "rrr",
                            tag: QuoteBlock,
                            trailling_whitespace: true,
                        },
                        Word {
                            word: "sss",
                            tag: QuoteBlock,
                        },
                        Word {
                            word: "",
                            tag: QuoteBlock,
                            trailling_whitespace: true,
                        },
                        Word {
                            word: "tt",
                            tag: QuoteBlock,
                        },
                    ],
                },
            ],
            links: [],
            footnotes: [],
        },
        Block {
            lines: [
                Line {
                    words: [
                        Word {
                            word: "",
                        },
                        Word {
                            word: "* ",
                            tag: ListItem,
                        },
                        Word {
                            word: "[x]",
                        },
                        Word {
                            word: "",
                            trailling_whitespace: true,
                        },
                        Word {
                            word: "done!",
                        },
                    ],
                },
                Line {
                    words: [
                        Word {
                            word: "  ",
                        },
                        Word {
                            word: "* ",
                            tag: ListItem,
                        },
                        Word {
                            word: "nested",
                            trailling_whitespace: true,
                        },
                        Word {
                            word: "list",
                        },
                    ],
                },
                Line {
                    words: [
                        Word {
                            word: "",
                        },
                        Word {
                            word: "* ",
                            tag: ListItem,
                        },
                        Word {
                            word: "[ ]",
                        },
                        Word {
                            word: "",
                            trailling_whitespace: true,
                        },
                        Word {
                            word: "undone",
                        },
                    ],
                },
                Line {
                    words: [
                        Word {
                            word: "  ",
                        },
                        Word {
                            word: "1. ",
                            tag: ListItemN(
                                1,
                            ),
                        },
                        Word {
                            word: "a",
                            style.add_modifier: ITALIC,
                        },
                    ],
                },
                Line {
                    words: [
                        Word {
                            word: "  ",
                        },
                        Word {
                            word: "2. ",
                            tag: ListItemN(
                                2,
                            ),
                        },
                        Word {
                            word: "`",
                            tag: InlineCode,
                        },
                        Word {
                            word: "b",
                            tag: InlineCode,
                        },
                        Word {
                            word: "`",
                            tag: InlineCode,
                        },
                    ],
                },
            ],
            links: [],
            footnotes: [],
        },
    ],
    links: Links {
        heading: [
            (
                1,
                "# h1 `code`\n",
                SelectedRegion {
                    row_start: 0,
                    row_end: 1,
                    col_start: 0,
                    col_end: 5,
                },
            ),
        ],
        links: [],
        footnotes: {},
    },
}
